roachpb/errors.go

```diff
@@ -172,6 +172,17 @@ func (e *Error) SetGoError(err error) {
 	}
 }
 
+// SetTxn sets the txn and resets the error message.
+func (e *Error) SetTxn(txn *Transaction) {
```

Comment:
Can we give the field a custom name in the proto file to ensure that all access goes through setter methods (and a new getter)?